crates/runtime/src/memory.rs

```diff
@@ -479,11 +643,13 @@ impl Memory {
         // calculation overflowed. This means that the `minimum` we're informing
         // the limiter is lossy and may not be 100% accurate, but for now the
         // expected uses of limiter means that's ok.
-        if !store.memory_growing(0, minimum.unwrap_or(absolute_max), maximum)? {
-            bail!(
-                "memory minimum size of {} pages exceeds memory limits",
-                plan.memory.minimum
-            );
+        if let Some(store) = store {
```

Comment:
This change is actually a bit worrisome to me because the purpose of this argument is to limit memory usage within a `Store`, but our current strategy of a per-`Store` memory limiter is not sufficient to protect against shared memories.

If `store` is `None` could this assert that the memory is shared? Additionally I think we'll need to keep a note on a tracking issue for shared memories to figure out how best to limit them since our current `ResourceLimiter` won't work